crates/assembly-syntax/src/ast/item/resolver.rs

```diff
@@ -0,0 +1,390 @@
+use alloc::{collections::BTreeMap, sync::Arc, vec::Vec};
+
+use miden_debug_types::{SourceSpan, Span, Spanned};
+
+use crate::{
+    Path, Word,
+    ast::{AliasTarget, Ident, ItemIndex},
+    diagnostics::{Diagnostic, miette},
+};
+
+use super::{GlobalItemIndex, ModuleIndex};
+
+/// Represents the result of resolving a symbol
+#[derive(Debug, Clone)]
+pub enum SymbolResolution {
+    /// The name was resolved to a definition in the same module at the given index
+    Local(Span<ItemIndex>),
+    /// The name was resolved to an path referring to an item exported from another module
+    External(Span<Arc<Path>>),
+    /// The name was resolved to a procedure MAST root
+    MastRoot(Span<Word>),
+    /// The name was resolved to a known definition in the same module at the given index
+    Exact {
+        gid: GlobalItemIndex,
+        path: Span<Arc<Path>>,
+    },
```

Comment:
Is the `path` pointing to the symbol? So the `gid` and `path` are equivalent representations of the symbol?

Also, the docstring is basically the same as `Local` - is this right?